Engine/Entities/Colonies/Colony.cs

```diff
@@ -0,0 +1,132 @@
+        {
+            _population = new List<IColonyMember>();
+            _stockpile = new ResourceDeposit();
+            _spawnCost = new ResourceDeposit();
+
+            SpawnMethod = spawnMethod;
+        }
+
+        /// <summary>
+        /// Current population of the colony.
+        /// This is basically the entities that have been spawned by the colony.
+        /// </summary>
+        public IList<IColonyMember> Population => _population.ToImmutableList();
+
+        /// <summary>
+        /// Current stockpile of the colony.
+        /// The resources that can be used to spawn more entities.
+        /// </summary>
+        public ResourceDeposit Stockpile => _stockpile;
+
+        /// <summary>
+        /// The cost in resources of spawning one entity.
+        /// </summary>
+        public ResourceDeposit SpawnCost => _spawnCost;
+
+        /// <summary>
+        /// Distance of spawning from the colony center.
+        /// </summary>
+        private float SpawnRadius { get; set; } = 1F;
+
+        /// <summary>
+        /// Where an entity should spawn.
+        /// </summary>
+        protected virtual Vector2 SpawnPosition
+        {
+            get
+            {
+                float angle = (float)new Random().NextDouble();
+                float x = MathF.Cos(angle);
+                float y = MathF.Sin(angle);
+
+                return Transform.Position + new Vector2(x, y) * SpawnRadius;
+            }
+        }
+
+        /// <summary>
+        /// Instructions to spawn an entity of the colony.
+        /// </summary>
+        public ColonySpawnMethod SpawnMethod { private get; set; }
+
+        /// <summary>
+        /// Spawn entities stopping when at count or when stockpile has not enough resources.
+        /// </summary>
+        /// <param name="count">Number of entities to spawn</param>
+        public virtual void Spawn(int count = 1)
+        {
+            for (int i = 0; i < count && HasEnoughResources(); i++)
+            {
+                ConsumeResources();
+
+                IColonyMember pop = SpawnMethod("", new Transform(SpawnPosition, 0, Vector2.One), World, this);
+                _population.Add(pop);
+                pop.Home = this;
+            }
+        }
+
+        /// <summary>
+        /// Checks if the colony has enough resources in its stockpile to spawn an entity.
+        /// </summary>
+        /// <returns>True if it can, false otherwise</returns>
+        protected bool HasEnoughResources()
+        {
+            foreach ((Resource resource, int cost) in _spawnCost.All)
+            {
+                if (_stockpile.All.ContainsKey(resource))
+                {
+                    if (_stockpile.All[resource] < cost) return false;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
```

Comment:
Does this take into account the fact that maybe an entity could use up a certain combination of resources (for example an ant could need 20µg of food and 5µg of water)? Should we even be checking for that?